cmd/minishift/cmd/root.go

```diff
@@ -213,14 +222,37 @@ func initializeProfile() string {
 		activeProfile string
 	)
 
+	// If `minishift profile set` or `minishift start --profile` command is used.
 	for i, arg := range os.Args {
+		// This will match if `--profile` flag is used
 		if arg == "--"+profileFlag {
+			// This make sure if user specify `--profile` flag without any name
+			// it should not panic with out of index error
+			if len(os.Args) <= i+1 {
+				break
```

Comment:
We do not need this check as Cobra/Viper takes care of this. In my testing I found this as dead code.